app/controllers/price_policies_controller.rb

```diff
@@ -33,28 +31,36 @@ def index
 
   # GET /facilities/:facility_id/{product_type}/:product_id/price_policies/new
   def new
-    @start_date = new_start_date
-    @expire_date = set_max_expire_date
+    # If there are active policies, start tomorrow. If none, start today
+    @start_date = Date.today + (active_policies? ? 1 : 0)
+
     @price_policies = PricePolicyBuilder.get_new_policies_based_on_most_recent(@product, @start_date)
     raise ActiveRecord::RecordNotFound if @price_policies.blank?
-    render "price_policies/new"
   end
 
   # POST /facilities/:facility_id/{product_type}/:product_id/price_policies
   def create
-    create_or_update(:new)
+    if update_policies_from_params!
+      redirect_to facility_product_price_policies_path, notice: text("create.success")
+    else
+      flash.now[:error] = text("errors.save")
+      render :new
+    end
   end
 
   # PUT /facilities/:facility_id/{product_type}/:product_id/price_policies/:id
   def update
-    create_or_update(:edit)
+    if update_policies_from_params!
```

Comment:
It seems strange to use a conditional with a persistence method that ends with a `!` here (because the Rails methods with names like that raise exceptions). I see that the culprit is `PricePolicyUpdater.update_all!` (which calls `save` in its implementation, and not `save!`), but perhaps we can rename `update_policies_from_params!` to not have a `!`?

Reply:
It calls yeah, you're right. I think the exclamations all the way down led me to use that here, but you're right that we don't need that. Will update.